test/asynchronous/test_cursor.py

```diff
@@ -1380,6 +1380,17 @@ async def test_getMore_does_not_send_readPreference(self):
         self.assertEqual("getMore", started[1].command_name)
         self.assertNotIn("$readPreference", started[1].command)
 
+    @async_client_context.require_sync
```

Comment:
This test should run on both async and sync.

Reply:
`__getitem__` is not supported for async cursors and is the only way to trigger `self._empty == True`.